db.go

```diff
@@ -460,6 +500,59 @@ func retentionCutoffDirs(dir string, mint int64) ([]string, error) {
 	return delDirs, nil
 }
 
+// maxBytesCutoffDirs returns all the oldest directories that would need to be
+// deleted to maintain the 'MaxBytes' option.
+func maxByteCutoffDirs(dir string, limit int64, metrics *dbMetrics) ([]string, error) {
+	df, err := fileutil.OpenDir(dir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "open directory")
+	}
+	defer df.Close()
+
+	dataDirSize, err := dirSize(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	delDirs := []string{}
+	if (dataDirSize <= limit) {
+		return delDirs, nil
```

Comment:
If the list is empty, you can just return `nil` here for it and define the variable `delDirs` further below where it's actually needed.